### ISSUE TITLE
[Ehancement] Compose iOS only Loads resources from main bundle

### ISSUE THREAD

**Issue by ricardocolombo** [JetBrains/compose-multiplatform#4234]
From many versions now, and including the next 1.6.0-beta, the resources on iOS are only loaded from the Main Bundle: 

https://github.com/JetBrains/compose-multiplatform/blob/c31c761e09212eaa13014f4d0d2a6516511f859a/components/resources/library/src/iosMain/kotlin/org/jetbrains/compose/resources/ResourceReader.ios.kt#L13-L14

This works fine if the integrator application includes the resources needed, but does not work at all if the idea is for a KMM-built framework (e.g. distributed via SPM) to provide UI out of the box, using those resources.
The Framework's resources live only inside the "Frameworks/_CustomLibrary_.framework/" folder, and thus, none of its resources can be accessed by Compose.

It would be crucial for further usage of Compose Multiplatform if those resources could actually be fetched from the library's resources, if the _painter_ would allow providing a root folder, or something in that direction.

Please let me know also if I'm mistaken and there's a direct way for Compose Multiplatform to access the resources files in a scenario like this for iOS.

Thank you.

**Comment by terrakok** [comment]
Yes. It is the current restriction. Kotlin Native distribution with resources is not supported yet.

**Comment by terrakok** [comment]
It should be a part of logic of the Kotlin gradle plugin, so could you file the issue here: https://kotl.in/issue

**Comment by ricardocolombo** [comment]
Thank you for the answer, @terrakok!

Even if the native distribution of resources is not supported by Kotlin, one can still provide those files by, e.g., packaging the SPM file manually and including the resource files. Those resource files that can be later simply be fetched with:
```
val bundle = NSBundle.bundleWithIdentifier("com.example.your_bundle") // framework bundle identifier
val filePath = bundle?.pathForResource(fileName, ofType = extension) ?: ""
```
This can be achieved without any problems, but the implementation on Compose to look for files only in the `NSBundle.mainBundle` imposes a hard restriction, preventing a library from consuming resources directly from its own bundle.

**Comment by terrakok** [comment]
The main scenario of the usage K/N-framework is a single framework per app. So, It doesn't seem a relevant problem at the moment, does it?

**Comment by terrakok** [comment]
I see the problem with a distribution of a common code as a XCFramework but, as I said, it is not supported yet. At the moment a final native compose app should be build all together from a kotlin code on a single machine

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4234">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
